homeassistant/components/remootio/cover.py

```diff
@@ -0,0 +1,234 @@
+        """Open the Remootio controlled garage door or gate."""
+        await self.__remootio_client.trigger_open()
+
+    def close_cover(self, **kwargs: Any) -> None:
+        """Close the Remootio controlled garage door or gate."""
+        self.hass.async_create_task(self.async_close_cover())
+
+    async def async_close_cover(self, **kwargs):
+        """Close the Remootio controlled garage door or gate."""
+        await self.__remootio_client.trigger_close()
+
+
+class RemootioCoverStateChangeListener(Listener[StateChange]):
+    """Listener to be invoked when Remootio controlled garage door or gate changes its state."""
+
+    __owner: RemootioCover
+
+    def __init__(self, owner: RemootioCover) -> None:
+        """Initialize an instance of this class."""
+        super().__init__()
+        self.__owner = owner
+
+    async def execute(self, client: RemootioClient, subject: StateChange) -> None:
+        """Execute this listener. Tell Home Assistant that the Remootio controlled garage door or gate has changed its state."""
+        _LOGGER.debug(
+            f"Telling Home Assistant that the Remootio controlled garage door or gate has changed its state. RemootioClientState [{client.state}] RemootioCoverEntityId [{self.__owner.entity_id}] RemootioCoverUniqueId [{self.__owner.unique_id}] RemootioCoverState [{self.__owner.state}]"
+        )
+        self.__owner.async_schedule_update_ha_state()
+
+
+class RemootioCoverEventListener(Listener[Event]):
+    """Listener to be invoked on an event sent by the Remmotio device."""
+
+    __owner: RemootioCover
+
+    def __init__(self, owner: RemootioCover) -> None:
+        """Initialize an instance of this class."""
+        super().__init__()
+        self.__owner = owner
+
+    async def execute(self, client: RemootioClient, subject: Event) -> None:
+        """Execute this listener. Fire events in Home Assistant based on events sent by the Remootio device.
+
+        Fires the event remootio_left_open in Home Assistant when the Remootio controlled garage door or gate has been left open.
+        As event data the serial number of the Remootio device (also the id of the entity which represents it in Home Assistant) and
+        the name of the entity which represents the Remootio device in Home Assistant will be passed to the fired event.
+        """
+        if subject.type == EventType.LEFT_OPEN:
+            event_type = f"{DOMAIN.lower()}_{subject.type.name.lower()}"
+
+            _LOGGER.debug(
+                f"Firing event. EvenType [{event_type}] RemootioCoverEntityId [{self.__owner.entity_id}] RemootioCoverUniqueId [{self.__owner.unique_id}]"
+            )
+
+            self.__owner.hass.bus.async_fire(
+                event_type,
+                {
+                    ED_ENTITY_ID: self.__owner.entity_id,
+                    ED_SERIAL_NUMBER: self.__owner.unique_id,
+                    ED_NAME: self.__owner.name,
+                },
```

Comment:
This should really be converted to DataUpdateCoordinator. With CoordinatorEntity  entities. This will schedule there own update once the coordinator updates.

Reply:
> This should really be converted to DataUpdateCoordinator. With CoordinatorEntity entities. This will schedule there own update once the coordinator updates.

I don't understand why I should use a DataUpdateCoordinator here?

After a time that can be configured via the Remootio app, the device triggers a `LEFT_OPEN` event if the controlled garage door or garden gate has been left open. This event is received by the `RemootioClient` and calls the listener `RemootioCoverEventListener` for it, which in turn triggers a `remootio_left_open` Home Assistant event.

Reply:
I'm not saying your solution doesnt work. But you add boilerplate signalling that the update coordinator reduces.

Reply:
The event is only fired once from the device if the garage door or garden gate has been left open.

Reply:
This code is more code compared to using a update coordinator.

Reply:
According to the [documentation](https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities), the `DataUpdateCoordinator` is intended for polling multiple entities for data, but this is about handling an event that fired the device.

Reply:
There is only one entity now. We are adding things like config and diagnostics as there own entities for other integrations, so it's likely there will be more.

Coordinator doesn't have to only be for polled entities. You can set it's data based on events too, which triggers entities linked to it to automatically update.

Reply:
I'll keep that in mind and adjust it if necessary.  At the moment I don't see any need for this.

Reply:
Ps. I will need some other eyes on this before merge. It's quite likely the comment will be the same.